src/parser.ts

```diff
@@ -400,22 +355,27 @@ export class Parser extends DiagnosticEmitter {
       }
     }
 
-    // check if this an `export default` declaration
-    if (defaultEnd && statement !== null) {
+    if (modifierFlags != 0 && statement !== null) {
       switch (statement.kind) {
         case NodeKind.ENUMDECLARATION:
         case NodeKind.FUNCTIONDECLARATION:
         case NodeKind.CLASSDECLARATION:
         case NodeKind.INTERFACEDECLARATION:
-        case NodeKind.NAMESPACEDECLARATION: {
-          return Node.createExportDefaultStatement(<DeclarationStatement>statement, tn.range(startPos, tn.pos));
-        }
-        default: {
+        case NodeKind.NAMESPACEDECLARATION:
+          if (flags & CommonFlags.DEFAULT)
+            return Node.createExportDefaultStatement(<DeclarationStatement>statement, tn.range(startPos, tn.pos));
```

Comment:
I believe this `tn.range(startPos, tn.pos)` is redundant and we can just use `statement.range`